executor/insert_common.go

```diff
@@ -1201,10 +1201,13 @@ func (e *InsertValues) batchCheckAndInsert(ctx context.Context, rows [][]types.D
 					if handle == nil {
 						continue
 					}
-					_, err = e.removeRow(ctx, txn, handle, r, true)
+					skip, err = e.removeRow(ctx, txn, handle, r, true)
 					if err != nil {
 						return err
 					}
+					if skip {
+						break
```

Comment:
`break` will skip `removeRow` for rest `r.uniqueKeys`. I'm not sure if this is expected. Please find some execution reviewers to check